### PR TITLE
feat: add callback for when carousel active id changes

### LINKED ISSUE #1825: Add slide navigation callback to/in Carousel
**What feature are you requesting?**
Need a callback prop on the Carousel for the parent to get notified when the user navigates back/forth between slides using click or keyboard. I need to be able to basically only need this for when the user initiates it (for example not when code triggers an active slide id change). So if you're exposing a generic callback for when next/prev is done, I need an argument to identify this is for an actual user initiated flip. I don't care about the programmatic change scenario, but if you decide to put it under the same callback, I need a way to distinguish between the two.

**What would be the benefit of adding this feature?**
MSN News Infopane experience's auto rotation feature. Currently I'm using a generic 'onclick' handler to stop the auto rotation, to indicate that the user has interacted with the carousel. But I will be missing things like the keyboard navigation, etc. 

So from the infopane perspective I need a way to start/resume from where the user left off, but if the user manually interacts with the carousel in the meantime, then I don't have a way to keep track of the *current slide id* to know which slide to flip to when the next auto-rotation timer fires.

**What solution would be ideal for you?**
A callback on the CarouselHandledProps to indicate that the user has navigated prev/next via mouse or keyboard (or any other mechanism that I'm unaware of).

**What alternatives have you considered?**
Resorting to using a blind 'onclick' handler to trigger the auto-rotation stop behavior.

**Is there any additional context for your request?**

**Attach screenshots or images add detail to your request:**

### PR DESCRIPTION
**Pull request by chrisdholt**
closes #1825 

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/en/contributing/standards) for this project.

### CODE REVIEW

**Comment by prabashn** [comment]
Does this callback get triggered whenever the carousel changes regardless of the source of the change (i.e., user clicks vs. programmatic changes?)

File: packages/fast-components-react-msft/src/carousel/carousel.spec.tsx [hunk 1]
```diff
@@ -236,6 +236,62 @@ describe("carousel", (): void => {
...
+
+        rendered
+            .find('[direction="previous"]')
+            .first()
+            .simulate("click");
+
+        expect(onActiveIdUpdateFn.mock.results[1].value).toBe("id01");
+
+        rendered
+            .find('[role="tab"]')
```
Comment (prabashn) [inline packages/fast-components-react-msft/src/carousel/carousel.spec.tsx]: Is this the little indicators? If not, consider adding a test for those as well.

Comment (chrisdholt) [reply packages/fast-components-react-msft/src/carousel/carousel.spec.tsx]: Yeah - this is the indicator. Checking click for each of them.

**Comment by chrisdholt** [comment]
> Does this callback get triggered whenever the carousel changes regardless of the source of the change (i.e., user clicks vs. programmatic changes?)

Yes, I believe this should trigger any time the activeID is changed regardless of input.

**Pull request merged**